framework/auth/core.py

```diff
@@ -843,6 +849,35 @@ def _get_unconfirmed_email_for_token(self, token):
 
         return verification['email']
 
+    def clean_email_verifications(self):
+        email_verifications = deepcopy(self.email_verifications)
+        for token in self.email_verifications:
+            try:
+                self.get_unconfirmed_email_for_token(token)
+            except (KeyError, ExpiredTokenError):
+                email_verifications.pop(token)
+        self.email_verifications = email_verifications
+
+    def unconfirmed_email_get(self):
```

Comment:
You've made a view into a method, so make sure to change the name and docstring accordingly (e.g. "methods: GET" no longer applies). This should probably be made a property, since it's a...property...of the user =).  

Whether it's a property or a getter method, this should not do any writes, i.e. clean_email_verifications should not be called here. Users of class will not expect `user.get_unconfirmed_emails()` (getter method) or `user.unconfirmed_emails` (property) to write to the database.